File: WebClient SDK/src/default/workflow/krProcess/krGlobalTileCommand.ts
import { IKrTileCommand } from './interfaces';
import { Guid } from 'tessa/platform';
import { IUIContext, LoadingOverlay, showConfirm, showNotEmpty } from 'tessa/ui';
import { KrProcessInstance, KrTileInfo, launchProcess } from 'tessa/workflow/krProcess';
import { ITile } from 'tessa/ui/tiles';

export class KrGlobalTileCommand implements IKrTileCommand {
  //#region ctor

  private constructor() {}

  //#endregion

  //#region instance

  private static _instance: KrGlobalTileCommand;

  public static get instance(): KrGlobalTileCommand {
    if (!KrGlobalTileCommand._instance) {
      KrGlobalTileCommand._instance = new KrGlobalTileCommand();
    }
    return KrGlobalTileCommand._instance;
  }

  //#endregion

  public async onClickAction(_context: IUIContext, _tile: ITile, tileInfo: KrTileInfo) {
    if (tileInfo.id === Guid.empty) {
      return;
    }

    if (tileInfo.askConfirmation) {
      const result = await showConfirm(tileInfo.confirmationMessage);
      if (!result) {
        return;
      }
    }

    await LoadingOverlay.instance.show(async () => {
      const process = KrProcessInstance.createWithParams({ processId: tileInfo.id });
      const result = await launchProcess(process, { raiseErrorWhenExecutionIsForbidden: true });
      if (result) {
        await showNotEmpty(result.validationResult.build());
      }
    });
  }
}

File: Source/Extensions/Tessa.Extensions.Default.Client/Workflow/KrProcess/KrGlobalTileCommand.cs
using System.Threading.Tasks;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.Localization;
using Tessa.UI;
using Tessa.UI.Tiles;

namespace Tessa.Extensions.Default.Client.Workflow.KrProcess
{
    /// <summary>
    /// Команда выполняющаяся при клике по глобальному тайлу подсистемы маршрутов.
    /// </summary>
    public sealed class KrGlobalTileCommand : IKrTileCommand
    {
        private readonly IKrProcessLauncher launcher;


        public KrGlobalTileCommand(
            IKrProcessLauncher launcher)
        {
            this.launcher = launcher;
        }


        /// <inheritdoc />
        public async Task OnClickAsync(
            IUIContext context,
            ITile tile,
            KrTileInfo tileInfo)
        {
            if (tileInfo.ID == default)
            {
                return;
            }

            if (tileInfo.AskConfirmation
                && !TessaDialog.Confirm(LocalizationManager.Format(tileInfo.ConfirmationMessage)))
            {
                return;
            }

            using (TessaSplash.Create("$KrButton_DefaultTileSplash"))
            {
                var process = KrProcessBuilder
                    .CreateProcess()
                    .SetProcess(tileInfo.ID)
                    .Build();

                var specificParameters = new KrProcessUILauncher.SpecificParameters
                {
                    RaiseErrorWhenExecutionIsForbidden = true
                };

                var result = await this.launcher.LaunchAsync(process, specificParameters: specificParameters);

                TessaDialog.ShowNotEmpty(result.ValidationResult);
            }
        }
    }
}